mmv1/products/dlp/api.yaml

```diff
@@ -1328,4 +1330,337 @@ objects:
+                                    - !ruby/object:Api::Type::Integer
+                                      name: seconds
+                                      description: Seconds of minutes of the time. Must normally be from 0 to 59. An API may allow the value 60 if it allows leap-seconds.
+                                    - !ruby/object:Api::Type::Integer
+                                      name: nanos
+                                      description: Fractions of seconds in nanoseconds. Must be from 0 to 999,999,999.
+                                - !ruby/object:Api::Type::NestedObject
+                                  name: dateValue
+                                  description: Represents a whole or partial calendar date.
+                                  properties:
+                                    - !ruby/object:Api::Type::Integer
+                                      name: year
+                                      description: Year of the date. Must be from 1 to 9999, or 0 to specify a date without a year.
+                                    - !ruby/object:Api::Type::Integer
+                                      name: month
+                                      description: Month of a year. Must be from 1 to 12, or 0 to specify a year without a month and day.
+                                    - !ruby/object:Api::Type::Integer
+                                      name: day
+                                      description: Day of a month. Must be from 1 to 31 and valid for the year and month, or 0 to specify a year by itself or a year and month where the day isn't significant.
+                                - !ruby/object:Api::Type::Enum
+                                  name: dayOfWeekValue
+                                  description: Represents a day of the week.
+                                  values:
+                                    - :MONDAY
+                                    - :TUESDAY
+                                    - :WEDNESDAY
+                                    - :THURSDAY
+                                    - :FRIDAY
+                                    - :SATURDAY
+                                    - :SUNDAY
+                        - !ruby/object:Api::Type::NestedObject
+                          name: redactConfig
+                          description: Redact
+                          properties: [] # Meant to be an empty object with no properties - see here : https://cloud.google.com/dlp/docs/reference/rest/v2/projects.deidentifyTemplates#redactconfig
+                          # The fields below allow POST requests to include the "redactConfig" transformation in the payload
+                          # The trade-off is that POST requests specifying another transformation type will include will include "redactConfig" : null in the payload
+                          send_empty_value: true
+                          allow_empty_object: true
```

Comment:
This field is represented as an empty object and has no properties that can be defined here. I benefitted from [this work](https://github.com/GoogleCloudPlatform/magic-modules/commit/520e78b5db2970a5b0bbbbd86427043842ad442b) that added the ability to use `properties: []`.

I need to set both `send_empty_value` and `allow_empty_object` so that POST requests can contain an empty object when making a new template with redact config in use:

```
{
	"deidentifyTemplate": {
		"deidentifyConfig": {
			"recordTransformations": {
				"fieldTransformations": [{
					...
					"primitiveTransformation": {
						"redactConfig": {} ,
					}
				}]
			}
		},
	...
	}
}
```

However this means that if you supply a different type of transformation then the payload will always contain `"redactConfig": null,` at the same nested level:

```
{
	"deidentifyTemplate": {
		"deidentifyConfig": {
			"recordTransformations": {
				"fieldTransformations": [{
					...
					"primitiveTransformation": {
						"redactConfig": null,
						"replaceConfig": {
							"newValue": {
								"stringValue": "foobar"
							}
						}
					}
				}]
			}
		},
		...
	}
}
```